components/camel-zendesk/src/main/java/org/apache/camel/component/zendesk/ZendeskConfiguration.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.zendesk;
+
+import java.util.Map;
+
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.component.zendesk.internal.ZendeskApiName;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriParams;
+import org.apache.camel.spi.UriPath;
+
+/**
+ * Component configuration for Zendesk component.
+ */
+@UriParams
+public class ZendeskConfiguration {
+
+    @UriPath
+    @Metadata(required = "true")
+    private String methodName;
+
+    @UriParam
+    private String serverUrl;
```

Comment:
Can we make configuring the server url on the component as well, then we can configure this globally and also using spring boot configuration. 

The same for the security settings below. Allow to configure on component also. And then you can overrride per endpoint if you want